src/components/TodoApp.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from 'react';
+import { TodoList } from './TodoList';
+import { TodosFilter } from './TodosFilter';
+
+export const TodoApp = React.memo(() => {
+  const [input, setInput] = useState('');
+  const [todos, setTodos] = useState([]);
+  const [allTodos, setAllTodos] = useState(false);
+  const [filter, setFilter] = useState('All');
+  const [filteredTodos, setFilteredTodos] = useState([]);
+
+  const onClickHandler = (e) => {
+    setInput(e.target.value);
+  };
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+
+    if (!input) {
+      return;
+    }
+
+    const id = +new Date();
+    const todo = {
+      id,
+      title: input,
+      completed: false,
+    };
+
+    setTodos(state => [...state, todo]);
+    setInput('');
+  };
+
+  const allCompletedHandler = () => {
+    setAllTodos(!allTodos);
+    setTodos(todos.map((todo) => {
+      if (allTodos) {
+        return {
+          ...todo,
+          completed: true,
+        };
+      }
+
+      return {
+        ...todo,
+        completed: false,
+      };
+    }));
+  };
+
+  const filterHandler = () => {
+    switch (filter) {
+      case 'Completed':
+        setFilteredTodos(todos.filter(todo => todo.completed));
+        break;
+
+      case 'Active':
+        setFilteredTodos(todos.filter(todo => !todo.completed));
+        break;
+
+      default:
+        setFilteredTodos(todos);
+        break;
+    }
+  };
+
+  const removeTodosHandler = () => {
+    setTodos(todos.filter(todo => !todo.completed));
+  };
+
+  const setToLocalStrg = () => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  };
+
+  const getLocalTodos = () => {
+    if (localStorage.getItem('todos') === null) {
+      localStorage.setItem('todos', JSON.stringify([]));
+    } else {
+      setTodos(JSON.parse(localStorage.getItem('todos')));
+    }
+  };
+
+  useEffect(() => {
+    getLocalTodos();
+  }, []);
+
+  useEffect(() => {
+    filterHandler();
+    setToLocalStrg();
+  }, [todos, filter]);
+
+  const activeTodos = [...filteredTodos]
+    .filter(todo => !todo.completed).length;
+
+  const completedTodos = [...filteredTodos]
+    .filter(todo => todo.completed).length;
```

Comment:
do it inside useMemo, and only if you're currently on active/completed page

Reply:
Don`t really get what you mean. I use these template values all the time in the footer for conditional rendering and showing a number of active todos.